flox-bash/lib/metadata.sh

```diff
@@ -490,7 +490,14 @@ function checkGhAuth {
 	while ! $_gh auth status -h $hostname >/dev/null 2>&1; do
 		initialGreeting
 		warn "Invoking 'gh auth login -h $hostname'"
-		$_gh auth login -h $hostname
+
+		# gh auth login will automatically add credential helpers to the users
+		# global git config.
+		# Since flox will set the git credential helper manually where its needed
+		# and we want to avoid writing user files, trick gh to modify a temporary,
+		# discarded file instead
+		stub_git_config_file="$(mkTempFile)"
+		GIT_CONFIG_GLOBAL="$stub_git_config_file" $_gh auth login -h $hostname
```

Comment:
We either need to use the `local` keyword before the definition of `stub_git_config_file` or just dispense with it altogether like this:

```suggestion
		GIT_CONFIG_GLOBAL="$(mkTempFile)" $_gh auth login -h $hostname
```

Reply:
🐚
🔥